hooks/useIsContract.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+
+import { ChainFactory } from "../lib/chainFactory";
+import { EvmClientFactory } from "../lib/evmClient";
+
+const contractCache = new Map<string, boolean>();
+
+export function useIsContract(address: string) {
```

Comment:
Why not do this server side? I think we can assume overlap in the calls between clients and the result is not relevant for only the client's session, but also other clients

Reply:
If we wanted to run this server side and have the responses cached we'd need to use something like Redis. Remember server side components are rendered in lambda functions that are being torn down after they've completed their work.